File: SnakePhone/Renderer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using SnakeModel;
using Android.Graphics;

namespace SnakeWear
{
    public class Renderer
    {
        private SnakeGame game;

        private float scale;
        private Paint head;
        private Paint body;
        private Paint background;
        private Paint backgroundStroke;
        private Paint food;
        private Paint info;
        private Paint gameOver;

        private float paddingX;
        private float paddingY;
        private int width;
        private int height;
        private int highscore;

        public Renderer(SnakeGame game, int width, int height, int highscore)
        {
            this.game = game;
            this.width = width;
            this.height = height;
            this.highscore = highscore;

            scale = (Math.Min(width, height) / (float)SnakeGame.SIZE) * ((float)Math.Cos(Math.PI / 4));

            paddingX = (width - scale * SnakeGame.SIZE) / 2;
            paddingY = (height - scale * SnakeGame.SIZE) / 2;

            head = new Paint();
            head.Color = Color.DarkOrange;
            head.SetStyle(Paint.Style.FillAndStroke);

            body = new Paint();
            body.Color = new Color(100, 100, 100);
            body.SetStyle(Paint.Style.FillAndStroke);

            background = new Paint();
            background.Color = new Color(16, 16, 16);
            background.SetStyle(Paint.Style.Fill);

            backgroundStroke = new Paint();
            backgroundStroke.Color = Color.DarkOrange;
            backgroundStroke.SetStyle(Paint.Style.Stroke);

            food = new Paint();
            food.Color = Color.Cyan;
            food.SetStyle(Paint.Style.FillAndStroke);

            info = new Paint();
            info.Color = Color.Wheat;
            info.TextSize = scale*2;
            info.TextAlign = Paint.Align.Center;
            info.SetStyle(Paint.Style.Fill);

            gameOver = new Paint();
            gameOver.Color = Color.DarkOrange;
            gameOver.SetStyle(Paint.Style.FillAndStroke);
            gameOver.TextSize = 4 * scale;
            gameOver.TextAlign = Paint.Align.Center;
        }

        public void Render(Canvas canvas)
        {
            if (!game.GameOver)
            {
                canvas.DrawColor(Color.Black);
                DrawBackground(background, canvas);
                DrawBackground(backgroundStroke, canvas);
                canvas.DrawRect(game.Snake.Head.X * scale + paddingX, game.Snake.Head.Y * scale + paddingY, game.Snake.Head.X * scale + paddingX + scale, game.Snake.Head.Y * scale + paddingY + scale, head);
                foreach (SnekPiece piece in game.Snake.Pieces)
                {
                    DrawSnekPiece(piece, canvas);
                }

                float foodX = game.FoodX * scale + paddingX;
                float foodY = game.FoodY * scale + paddingY;
                canvas.DrawRect(foodX, foodY, foodX + scale, foodY + scale, food); //food
                DrawScore(canvas);
            }
            else
            {
                ShowGameOver(canvas);
            }
        }

        public int GetGameX(int x)
        {
            return (int)((x - paddingX) / scale);
        }

        public int GetGameY(int y)
        {
            return (int)((y - paddingY) / scale);
        }

        private void DrawSnekPiece(SnekPiece piece, Canvas canvas)
        {
            canvas.DrawRect(piece.X * scale + paddingX, piece.Y * scale + paddingY, piece.X * scale + paddingX + scale, piece.Y * scale + paddingY + scale, body);
        }

        private void DrawScore(Canvas canvas)
        {
            canvas.DrawText("Highest: " + GetHighscoreString(), width/2, paddingY - scale/2, info);
            canvas.DrawText("Current: " + GetScoreString(), width/2, paddingY + SnakeGame.SIZE*scale + scale*2, info);
        }

        private string GetScoreString()
        {
            return (1000 + game.Score).ToString().Substring(1);
        }

        private string GetHighscoreString()
        {
            return (1000 + Math.Max(highscore, game.Score)).ToString().Substring(1);
        }

        private void DrawBackground(Paint paint, Canvas canvas)
        {
            canvas.DrawRect(paddingX, paddingY, paddingX + SnakeGame.SIZE * scale, paddingY + SnakeGame.SIZE * scale, paint);
        }

        private void ShowGameOver(Canvas canvas)
        {
            string gameover = "Game over!";
            if (game.Score > highscore) gameover = "New highscore!";

            canvas.DrawPaint(background);
            gameOver.TextSize = 4 * scale;
            canvas.DrawText(gameover, width/2, height/2 - scale*2, gameOver);
            gameOver.TextSize = 3 * scale;
            canvas.DrawText("Your score: " + GetScoreString(), width / 2, height / 2 + scale * 2, gameOver);
            gameOver.TextSize = 2 * scale;
            canvas.DrawText("Tap to continue...", width / 2, height / 2 + scale * 8, gameOver);
        }
    }
}

File: android/snakewear/mobile/src/main/java/pontus/wearsnake/Renderer.java
package pontus.wearsnake;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;

import pontus.wearsnake.model.SnakeGame;
import pontus.wearsnake.model.SnekPiece;

/**
 * Created by Pontus on 2018-01-05.
 */

public class Renderer {
    private final int DARK_ORANGE = Color.parseColor("#ee7600");
    private final int GRAY = Color.parseColor("#646464");
    private final int DARK_GRAY = Color.parseColor("#101010");
    private final int CYAN = Color.parseColor("#00ffff");
    private final int WHEAT = Color.parseColor("#f5deb3");

    private SnakeGame game;

    private float scale;

    private Paint paint;

    private float paddingX;
    private float paddingY;
    private int width;
    private int height;
    private int highscore;

    public Renderer(SnakeGame game, int width, int height, int highscore) {
        this.game = game;
        this.width = width;
        this.height = height;
        this.highscore = highscore;

        scale = (Math.min(width, height) / SnakeGame.SIZE) * (float) (Math.cos(Math.PI / 4));

        paddingX = (width - scale * SnakeGame.SIZE) / 2;
        paddingY = (height - scale * SnakeGame.SIZE) / 2;

        paint = new Paint();
    }

    public void render(Canvas canvas) {
        if (!game.isGameOver()) {
            canvas.drawRGB(0,0,0);

            paint.setColor(DARK_GRAY);
            paint.setStyle(Paint.Style.FILL_AND_STROKE);
            drawBackground(paint, canvas);
            paint.setColor(DARK_ORANGE);
            paint.setStyle(Paint.Style.STROKE);
            drawBackground(paint, canvas);

            drawHead(canvas);
            paint.setColor(GRAY);
            paint.setStyle(Paint.Style.FILL_AND_STROKE);
            for (SnekPiece piece : game.getSnake().getBody()) {
                drawSnekPiece(piece, canvas);
            }

            float foodX = game.getFoodX() * scale + paddingX;
            float foodY = game.getFoodY() * scale + paddingY;
            paint.setColor(CYAN);
            paint.setStyle(Paint.Style.FILL_AND_STROKE);
            canvas.drawRect(foodX, foodY, foodX + scale, foodY + scale, paint); //food
            drawScore(canvas);
        } else {
            showGameOver(canvas);
        }
    }

    public int getGameX(int x) {
        return (int) ((x - paddingX) / scale);
    }

    public int getGameY(int y) {
        return (int) ((y - paddingY) / scale);
    }


    private void drawHead(Canvas canvas) {
        paint.setColor(DARK_ORANGE);
        paint.setStyle(Paint.Style.FILL_AND_STROKE);
        canvas.drawRect(
                game.getSnake().getHead().getX() * scale + paddingX,
                game.getSnake().getHead().getY() * scale + paddingY,
                game.getSnake().getHead().getX() * scale + paddingX + scale,
                game.getSnake().getHead().getY() * scale + paddingY + scale,
                paint);
    }

    private void drawSnekPiece(SnekPiece piece, Canvas canvas) {
        canvas.drawRect(
                piece.getX() * scale + paddingX,
                piece.getY() * scale + paddingY,
                piece.getX() * scale + paddingX + scale,
                piece.getY() * scale + paddingY + scale,
                paint);
    }

    private void drawBackground(Paint paint, Canvas canvas) {
        canvas.drawRect(paddingX, paddingY, paddingX + SnakeGame.SIZE * scale, paddingY + SnakeGame.SIZE * scale, paint);
    }

    private void showGameOver(Canvas canvas) {
        String gameover = "Game over!";
        if (game.getScore() > highscore) gameover = "New high score!";

        paint.setColor(DARK_ORANGE);
        paint.setStyle(Paint.Style.FILL_AND_STROKE);

        canvas.drawColor(DARK_GRAY);
        paint.setTextSize(3 * scale);
        canvas.drawText(gameover, width / 2, height / 2 - scale * 2, paint);
        paint.setTextSize(2 * scale);
        canvas.drawText("Your score: " + getScoreString(), width / 2, height / 2 + scale * 2, paint);
        paint.setTextSize(scale);
        paint.setColor(WHEAT);
        canvas.drawText("Tap to continue...", width / 2, height / 2 + scale * 8, paint);
    }

    private void drawScore(Canvas canvas) {
        paint.setColor(WHEAT);
        paint.setTextSize(scale * 2);
        paint.setTextAlign(Paint.Align.CENTER);
        paint.setStyle(Paint.Style.FILL);
        canvas.drawText("Highest: " + getHighscoreString(), width / 2, paddingY - scale / 2, paint);
        canvas.drawText("Current: " + getScoreString(), width / 2, paddingY + SnakeGame.SIZE * scale + scale * 2, paint);
    }

    private String getScoreString() {
        return String.valueOf(1000 + game.getScore()).substring(1);
    }

    private String getHighscoreString() {
        return String.valueOf(1000 + Math.max(highscore, game.getScore())).substring(1);
    }
}
